content/200-concepts/100-components/02-prisma-client/260-client-extensions/100-model.mdx

```diff
@@ -132,4 +132,88 @@ prisma.$extends({
 })
 ```
 
-You cannot use `Prisma.getExtensionContext(this)` inside an [`$allModels`](#add-a-custom-method-to-all-models-in-your-schema) structure.
+## Get the current model name at runtime
+
+<Admonition type = "info">
+
+This feature is available from version 4.9.0.
+
+</Admonition>
+
+You can get the name of the current model at runtime with `Prisma.getExtensionContext(this).name`. You might use this to write out the model name to a log, to send the name to another service, or to branch your code based on the model.
+
+For example:
+
+```ts
+// `ctx` refers to the current model
+const ctx = Prisma.getExtensionContext(this)
+
+// `ctx.name` returns the name of the current model
+console.log(ctx.name)
+```
+
+## Advanced type safety: improve the type safety and developer experience of your extensions
+
+<Admonition type = "info">
+
+This feature is available from version 4.9.0.
+
+</Admonition>
+
+To help you create highly type-safe extensions, Prisma Client provides a set of type utilities that tap into input and output types. They are fully dynamic, which means that they adapt to any given model and schema. You can use them to improve the auto-completion and developer exprience in your [shared extensions](/concepts/components/prisma-client/client-extensions/shared-extensions).
```

Comment:
Why is this on this page and not "shared extensions" then? Does it also have other use cases? Then we should list those I guess to avoid this confusion.

Reply:
I guess because this is used when you add things via `model` - so maybe the half sentence here missing is that this helps you to improve auto completion and dev ex of your added model methods, which especially makes sense when you want to share your extension.

Reply:
Yes, In general what works for a share extension also works for a local extension, so we kept it here.

Reply:
Good point Jan - I'll add that bit of information

Reply:
I've also changed the section heading to make it clearer that this section relates to custom model methods.